src/main/java/org/elasticsearch/plugin/keycloak/KeycloakSecurityPlugin.java

```diff
@@ -15,17 +15,28 @@
 
 package org.elasticsearch.plugin.keycloak;
 
-import org.elasticsearch.plugin.keycloak.realm.BearerToken;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.elasticsearch.common.settings.Setting;
+import org.elasticsearch.plugin.keycloak.realm.KeycloakRealm;
 import org.elasticsearch.plugins.ActionPlugin;
 import org.elasticsearch.plugins.Plugin;
+import org.elasticsearch.xpack.core.security.authc.RealmSettings;
 
-import java.util.Collection;
-import java.util.Collections;
+import java.util.ArrayList;
+import java.util.List;
 
 public class KeycloakSecurityPlugin extends Plugin implements ActionPlugin {
+    protected final Logger logger = LogManager.getLogger(this.getClass());
 
     @Override
-    public Collection<String> getRestHeaders() {
```

Comment:
These serve two different purposes.  getRestHeaders assigns any new rest headers that are required that Elastic normally wouldn't.  With ES 7, elastic introduced the BEARER_AUTH_HEADER as an acceptable header.  Because it was already defined, this caused the plugin to blow up upon install.

The getSettings function is a new item introduced in ES7.  This registers a new type of configuration that is acceptable to include in the YAML.  Specifically, the YAML to register the new type of authentication mechanism.  Elastic will fail to start if it detects a configuration in the YAML that is not registered.